pkg/actions/addon/addon.go

```diff
@@ -37,9 +44,45 @@ func New(clusterConfig *api.ClusterConfig, eksAPI eksiface.EKSAPI, stackManager
 		oidcManager:   oidcManager,
 		stackManager:  stackManager,
 		clientSet:     clientSet,
+		timeout:       5 * time.Minute,
 	}, nil
 }
 
+func (a *Manager) waitForAddonToBeActive(addon *api.Addon) error {
+	var out *awseks.DescribeAddonOutput
+	var err error
+	operation := func() (bool, error) {
+		out, err = a.eksAPI.DescribeAddon(&awseks.DescribeAddonInput{
+			ClusterName: &a.clusterConfig.Metadata.Name,
+			AddonName:   &addon.Name,
+		})
+		if err != nil {
+			return false, err
+		}
+		if *out.Addon.Status == awseks.AddonStatusActive {
+			return true, nil
+		}
+		return false, nil
+	}
+
+	w := waiter.Waiter{
+		Operation: operation,
+		NextDelay: func(_ int) time.Duration {
+			return a.timeout / 10
```

Comment:
The interval value shouldn't depend on the timeout value.  I'd prefer using a separate variable that's optionally configurable.

Reply:
why can't the re-queue period rely on on a fraction of the timeout? Whats the default behaviour if no `nextdelay` is set?

Reply:
>why can't the re-queue period rely on on a fraction of the timeout?

Because that can make the command wait longer than is required.

>Whats the default behaviour if no nextdelay is set?

There's no default behaviour, the waiter will always wait for the duration returned by `NextDelay`.